src/Merge/ExtraPackage.php

```diff
@@ -361,10 +362,30 @@ protected function mergeConflicts(RootPackage $root)
      */
     protected function mergeReplaces(RootPackage $root)
     {
-        $replaces = $this->package->getReplaces();
+        $rootReplaces = $root->getReplaces();
+
+        $replaces = array();
+        foreach ($this->package->getReplaces() as $name => $link) {
+            $constraint = $link->getConstraint();
+            $prettyConstraint = $link->getPrettyConstraint();
+
+            if ($prettyConstraint === 'self.version') {
+                $source = $rootReplaces[$link->getSource()];
```

Comment:
This will only work if the root package declares the same package replacement. Shouldn't this work like `AliasPackage::replaceSelfVersionDependencies()` instead and replace with a `new Constraint('=', $root->getVersion())` and `$root->getPrettyVersion()`?